### ISSUE TITLE
Redesign modify tags dialog

### ISSUE THREAD

**Issue by diegogangl** [getting-things-gnome/gtg#278]
The modify tags has a few issues. It forces the user to use a mini-language, and it's not possible to see _which_ tags you are modifying.

It would be better to have a listbox with all the tags, each row with their name and a dropdown/popup menu beside it. The dropdown menu would have three options: Don't change | Remove | Add to all tasks. Then at the end of the task a button to add another tag to all tasks.

Roughly similar to the Input Source list in the "Region and Language" panel of Gnome Settings.

Also rename to "Change tags" since we are not modifying the tags themselves, just changing which ones apply to the task.

**Comment by nekohayo** [comment]
Is this a duplicate of #276?

**Comment by diegogangl** [comment]
Nope, that's the tag editor (rename, color, etc.) This is the `CTRL+T` dialog

**Comment by diegogangl** [comment]
Glade + Inkscape mockup
![mockup_change_tags](https://user-images.githubusercontent.com/264829/80266296-79233b80-8671-11ea-9752-9c6aabd93d71.png)

**Comment by nekohayo** [comment]
While this "visually looks nice", I'm a bit worried about the efficiency and usability of it. Namely, how you deal with representing non-changes. The existing dialog lets you affect a bunch of tasks _regardless of whatever tags they may or may not individually have_ and then add and remove tags with _no risk for error_. That's really hard to represent with only the checkboxes' "On/Off/Inconsistent" three states. What if some tasks have unrelated tag foo but some other tasks don't have that tag, how do you represent that and how do you ensure the user can't screw it up?

Also, a cool feature of the existing implementation is that it remembers, from one instanciation of the dialog to another, what was the last used batch tagging string, so that I can do multiple batches of the same operation one after another if I "forgot" something.

**Comment by diegogangl** [comment]
I don't understand the risk of error. The functionality is essentially the same as the string, except for instantiation thing. 

> That's really hard to represent with only the checkboxes' "On/Off/Inconsistent" three states. What if some tasks have unrelated tag foo but some other tasks don't have that tag, how do you represent that and how do you ensure the user can't screw it up?

You represent it with the in-between state. In both the string dialog and this dialog the user can only do two things:  

1. Add a new tag to all selected tasks
2. Remove an existing tag from all selected tasks

In the string this would be typing the name of the tag to add, and !name to remove. In the dialog you would use the `+` button to add, and un-check a tag to remove. When you un-check the tag gets dimmer to help the user understand it's being removed. We could add a small message below the row saying "This tag will be removed" if that's not clear enough.

Likewise we could add a small tip below the name of the tag if it's in the in-between state: "Only set for some tasks".

**Comment by nekohayo** [comment]
> When you un-check the tag gets dimmer to help the user understand it's being removed. 

Using some markup to make the text struckthrough might be more obvious than just fading it out a bit...

> You represent it with the in-between state.

Hmm, I found something that might screw up your plans: I ran `gtk3-widget-factory` and tried clicking on the "inconsistent" checkbox there, and it refuses to change to Yes/No. Using the GTK Inspector, I changed its "inconsistent" property to False and then it happily accepted user input. But as soon as you set it back to True, it ignores all user input. So unless this is a bug in GTK or some specific weirdness in the way gtk3-widget-factory was made, it seems it would actually be impossible for us to use this for that purpose.

Even if it was possible... I'm a bit wary of the proposed change even though it looks beautiful visually,  because it really reduces the _efficiency_ of using the batch tagger:

* I no longer can reuse the same operation across multiple times I spawn that dialog (which is something I use surprisingly often)
* I can't use just the keyboard and now would have to use a bunch of mouse clicks. Instead of "ctrl+T, enter" (to redo the previous operation) or "ctrl+T, type some stuff in under 2 seconds, enter" I would now have to do "ctrl+T; click click click to deselect some of the existing tags, click to show the add tags dialog, click click click to add some new tags, click OK (or hit Enter) to dismiss the newly created dialog, Enter"
* If the user has many tags (I personally have 35 of them) it will have to make them scroll in whatever "+" tag picker you spawn, unless you arrange the tags in big alphabetical grid, and I'd like us to [avoid scrolling](https://fortintam.com/blog/dont-make-me-scroll/) at all costs.
* The existing dialog had autocompletion that works pretty well (although it could be improved). Unless you add a filtering search on top of it (it which point you might as well be retaining the existing text entry, but having both creates a new set of problems), the user must now spend time searching visually in an array of tags to find what widget to click on to get what they want. The visual search would slow me down a bit, I think.

I'm wondering if there's a way to preserve those speed advantages from the previous approach. My life would probably not grind to a halt, but I think I (and possibly other users) would find the new approach a bit less pleasant.

That said, the one advantage (besides gnomeish looks) I see with your proposal is that it makes this dialog touch-friendly. But then we haven't made touch-friendly multi-selections in the main window's task browser treeview, though, so unless that also happens then that point does not yet have significant weight.
